pinot-core/src/main/java/org/apache/pinot/core/common/datablock/DataBlockBuilder.java

```diff
@@ -84,15 +95,36 @@ public static RowDataBlock buildFromRows(List<Object[]> rows, DataSchema dataSch
       Object[] row = rows.get(rowId);
       for (int colId = 0; colId < numColumns; colId++) {
         Object value = row[colId];
+        ColumnDataType storedType = storedTypes[colId];
+
+        if (storedType == ColumnDataType.OBJECT) {
```

Comment:
move this part to switch `case OBJECT` is clear?

Reply:
This is intentional because we want to skip the step of setting `nullBitmaps`